### PR TITLE
Revert "chore(ui): Unify the logic for timeline item insertions"

### PR DESCRIPTION
**Pull request by poljar**
This seems to cause incorrect ordering under some more complex insertion scenarios which are not fully understood yet.

This is the likely culprit for: https://github.com/element-hq/element-x-ios/issues/3583.

A second commit, containing doc updates, was reverted as well as the culprit wouldn't revert cleanly on its own.

### CODE REVIEW

**Review by stefanceriu — APPROVED** [review]
Tested it on Element X iOS where we could originally easily reproduce the issue and I can't reproduce it anymore.

File: crates/matrix-sdk-ui/src/timeline/event_handler.rs [hunk 1]
```diff
@@ -1085,71 +1085,88 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
...
-                txn_id,
-                event_id,
-                ..
+
+            Flow::Remote {
+                position: TimelineItemPosition::End { .. }, txn_id, event_id, ..
             } => {
-                // This block tries to find duplicated events.
-
-                let removed_event_item_id = {
```
Comment (Hywan) [inline crates/matrix-sdk-ui/src/timeline/event_handler.rs:1102]: Can you run this block only if `position` is `TimelineItemPosition::End` please?

Comment (poljar) [reply crates/matrix-sdk-ui/src/timeline/event_handler.rs:1102]: What do you mean? Are you trying to find a fix for https://github.com/matrix-org/matrix-rust-sdk/pull/4378/commits/3b28809c354ca27cff95875cb0d420d2914616e7?

Comment (poljar) [reply crates/matrix-sdk-ui/src/timeline/event_handler.rs:1102]: Status update, yes he was trying to find a fix, but people want to do a release so we'll merge the revert and do the proper fix afterwards with a regression test.

**Review by Hywan — APPROVED** [review]

**Pull request merged**